src/main/java/io/vertx/proton/impl/ProtonConnectionImpl.java

```diff
@@ -371,12 +367,14 @@ public void fireDisconnect() {
     }
   }
 
-  void bindClient(NetClient client, NetSocket socket, ProtonSaslClientAuthenticatorImpl authenticator) {
+  void bindClient(NetClient client, NetSocket socket, ProtonSaslClientAuthenticatorImpl authenticator, ProtonClientOptions options) {
     transport = new ProtonTransport(connection, vertx, client, socket, authenticator);
+    transport.setIdleTimeout(options.getHeartbeat() * 2);
```

Comment:
In my opinion there are already too much parameter in the ProtonTransport class. 
I agree on having only one ProtonTransportOptions parameter for that.